Plugin/Http/Context.php

```diff
@@ -0,0 +1,188 @@
+    private $nostoHelperAccount;
+
+    /** @var NostoCmpHelperData  */
+    private $nostoCmpHelperData;
+
+    /** @var Store */
+    private $store;
+
+    public function __construct(
+        Session $customerSession,
+        CookieManagerInterface $cookieManager,
+        CategoryFactory $categoryFactory,
+        CategoryBuilder $categoryBuilder,
+        StoreManagerInterface $storeManager,
+        NostoHelperAccount $nostoHelperAccount,
+        NostoCmpHelperData $nostoCmpHelperData,
+        Http $request
+    ) {
+        $this->customerSession = $customerSession;
+        $this->cookieManager = $cookieManager;
+        $this->categoryFactory = $categoryFactory;
+        $this->categoryBuilder = $categoryBuilder;
+        $this->storeManager = $storeManager;
+        $this->nostoHelperAccount = $nostoHelperAccount;
+        $this->nostoCmpHelperData = $nostoCmpHelperData;
+        $this->request = $request;
+    }
+
+    /**
+     * @param MagentoContext $subject
+     * @return MagentoContext
+     */
+    function beforeGetVaryString(MagentoContext $subject)
+    {
+        if ($this->isCategoryPage() &&
+            $this->request->getParam(ParamResolver::DEFAULT_SORTING_ORDER_PARAM) &&
+            $this->request->getParam(ParamResolver::DEFAULT_SORTING_ORDER_PARAM) === NostoHelperSorting::NOSTO_PERSONALIZED_KEY &&
+            $this->nostoHelperAccount->nostoInstalledAndEnabled($this->store) &&
+            $this->nostoCmpHelperData->isCategorySortingEnabled($this->store)) {
+
+            $variation = $this->getSegmentFromCookie();
+            $subject->setValue('CONTEXT_NOSTO', $variation, $defaultValue = "");
+        }
+            return $subject;
+    }
+
+    /**
+     * Get segment id from cookie
+     * @return string
+     */
+    private function getSegmentFromCookie() {
+        //Read cookie
+        $cookie = $this->cookieManager->getCookie(SegmentMapping::COOKIE_NAME);
+
+        //Parse value
+        $stdClass = json_decode($cookie);
+        $segmentMap = get_object_vars($stdClass);
+
+        //Check if current category is part of segment mapping
+        if (array_key_exists($this->categoryString, $segmentMap) &&
+            property_exists($segmentMap[$this->categoryString] ,'segmentId')) {
+            $segment = $segmentMap[$this->categoryString];
+            if (property_exists($segment, 'variation')) {
+                return $segment->segmentId . '-' . $segment->variation;
+            }
+            return $segment->segmentId;
+        }
+        return '';
+    }
+
+    /**
+     * Checks if the current page is a category page
+     * @return bool
+     */
+    private function isCategoryPage() {
+        $category = $this->getCategory();
+        if ($category) {
+            $this->store = $this->storeManager->getStore();
+            $this->categoryString = strtolower(
+                $this->categoryBuilder->getCategory($category, $this->store)
+            );
```

Comment:
I don't think to set a class var inside a checker method is very intuitive. I'd extract this into `setCategoryString`